# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/test/gtest/APZTestCommon.h
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 gfx/layers/apz/test/gtest/APZTestCommon.h
--- a/gfx/layers/apz/test/gtest/APZTestCommon.h	Fri Mar 22 18:28:31 2019 +0000
+++ b/gfx/layers/apz/test/gtest/APZTestCommon.h	Fri Mar 22 18:28:42 2019 +0000
@@ -221,9 +221,21 @@
    **/
   void CancelAnimation() { EXPECT_TRUE(false); }
 
+  using APZCTreeManager::SetTargetAPZC;  // silence clang warning about overload
+  void SetTargetAPZC(uint64_t aInputBlockId,
+                     const nsTArray<ScrollableLayerGuid>& aTargets) {
+    nsTArray<SLGuidAndRenderRoot> wrapped;
+    for (const ScrollableLayerGuid& target : aTargets) {
+      wrapped.AppendElement(
+          SLGuidAndRenderRoot(target, wr::RenderRoot::Default));
+    }
+    this->SetTargetAPZC(aInputBlockId, wrapped);
+  }
+
  protected:
-  AsyncPanZoomController* NewAPZCInstance(
-      LayersId aLayersId, GeckoContentController* aController) override;
+  AsyncPanZoomController* NewAPZCInstance(LayersId aLayersId,
+                                          GeckoContentController* aController,
+                                          wr::RenderRoot aRenderRoot) override;
 
   TimeStamp GetFrameTime() override { return mcc->Time(); }
 
@@ -236,9 +248,11 @@
   TestAsyncPanZoomController(LayersId aLayersId,
                              MockContentControllerDelayed* aMcc,
                              TestAPZCTreeManager* aTreeManager,
+                             wr::RenderRoot aRenderRoot,
                              GestureBehavior aBehavior = DEFAULT_GESTURES)
       : AsyncPanZoomController(aLayersId, aTreeManager,
-                               aTreeManager->GetInputQueue(), aMcc, aBehavior),
+                               aTreeManager->GetInputQueue(), aMcc, aRenderRoot,
+                               aBehavior),
         mWaitForMainThread(false),
         mcc(aMcc) {}
 
@@ -873,11 +887,13 @@
 }
 
 AsyncPanZoomController* TestAPZCTreeManager::NewAPZCInstance(
-    LayersId aLayersId, GeckoContentController* aController) {
+    LayersId aLayersId, GeckoContentController* aController,
+    wr::RenderRoot aRenderRoot) {
   MockContentControllerDelayed* mcc =
       static_cast<MockContentControllerDelayed*>(aController);
   return new TestAsyncPanZoomController(
-      aLayersId, mcc, this, AsyncPanZoomController::USE_GESTURE_DETECTOR);
+      aLayersId, mcc, this, aRenderRoot,
+      AsyncPanZoomController::USE_GESTURE_DETECTOR);
 }
 
 inline FrameMetrics TestFrameMetrics() {